upack/src/upack/MsgPackReader.scala

```diff
@@ -98,7 +98,9 @@ abstract class BaseMsgPackReader extends upickle.core.BufferingByteParser{
   }
   def parseExt[T](n: Int, visitor: Visitor[_, T]) = {
     val (arr, i, j) = sliceArr(index + 1, n)
-    visitor.visitExt(getByteSafe(index), arr, i, j, index)
+    val res = visitor.visitExt(getByteSafe(index), arr, i, j, index)
+    index += n + 1
```

Comment:
This looks important!

If I'm reading correctly, looks like `n` is the array length, and the `+ 1` is the tag/ext-type byte. Seems like this fix should cover the fixext types as well. 👍